src/services/Services.ts

```diff
@@ -88,14 +89,15 @@ export default class Services {
    * This is a factory method to create Service container.
    * @param repositories Repository container.
    * @param config Instance of config.
+   * @param gatewayAddresses GatewayAddresses object.
    */
-  public static create(repositories: Repositories, config: Config): Services {
+  public static create(repositories: Repositories, config: Config, gatewayAddresses: GatewayAddresses): Services {
```

Comment:
We don't need to have gatewayAddresses as extra argument.
it can be wrapped under config object. config.mosaic can be removed